### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by domenicoblanco**
Added many unit tests and increased code coverage to 100%, also there is a little fix in an async function

### CODE REVIEW

File: tests/main_test.py [hunk 1]
```diff
@@ -4,47 +4,92 @@
...
+        'mock_func': ['get_users', 'run'],
+        'mock_ret': [[], None],
+    },
+    {
+        'func': main.main,
+        'expected_res': None,
+        'arg': tuple(),
+        'mock_obj': [main],
+        'mock_func': ['handle_urls'],
+        'mock_ret': [None],
```
Comment (Valeri0p) [inline tests/main_test.py]: Unneeded comma at the end of the last four test entries.

Comment (domenicoblanco) [reply tests/main_test.py]: Thanks, I didn't notice

**Comment by Valeri0p** [comment]
There is probably a mismatch between `mock_obj, mock_func and mock_ret` argument numbers in various tests.

**Review by Valeri0p — APPROVED** [review]

**Comment by Valeri0p** [comment]
I think we might have solved #17 :)

File: tests/main_test.py [hunk 2]
```diff
@@ -4,47 +4,124 @@
...
-        'func': main.check_ping,
-        'expected_res': False,
-        'arg': ('wrongurlpage.com',),
-        'is_async': False
+        'func': main.handle_urls,
+        'expected_res': None,
+        'arg': ('http://example.com', 'get'),
+        'mock_obj': [main,main],
+        'mock_func': ['run','handle_communication'],
+        'mock_ret': [False,None]
```
Comment (domenicoblanco) [inline tests/main_test.py]: Some spaces would increase readability

**Pull request merged**